stan/math/opencl/multiply.hpp

```diff
@@ -37,6 +37,28 @@ inline auto multiply(const matrix_cl& A, const matrix_cl& B) {
     temp.zeros();
     return temp;
   }
+  if (A.rows() == 1 && triangular_view_A == TriangularViewCL::Entire
+      && triangular_view_B == TriangularViewCL::Entire) {
+    try {
+      opencl_kernels::row_vector_matrix_multiply(
+          cl::NDRange(temp.cols() * 64), cl::NDRange(64), A.buffer(),
+          B.buffer(), temp.buffer(), B.rows(), B.cols());
+    } catch (cl::Error& e) {
+      check_opencl_error("multiply", e);
```

Comment:
Change the multiply to row_vector x matrix or something more specific then just multiply.